resources/hab_build.rb

```diff
@@ -20,20 +22,28 @@
     command "sudo -E #{hab_binary} studio" \
             " -r #{hab_studio_path}" \
             " build #{plan_dir}"
-    environment('TERM' => 'vt100',
-                'HAB_ORIGIN' => origin,
-                'HAB_NONINTERACTIVE' => 'true')
+    environment({ 'TERM' => 'vt100',
+                  'HAB_ORIGIN' => origin,
+                  'HAB_NONINTERACTIVE' => 'true' }.merge(
+                    new_resource.environment
```

Comment:
@smith is right, if you want the user to be able to override the defaults it needs to go in the order he had it.